### PR TITLE
Merge github.com:nomis52/ja-rule into licence

### PR DESCRIPTION
**Pull request by nomis52**

### CODE REVIEW

File: tests/scripts/enforce_licence.py [hunk 1]
```diff
@@ -172,10 +172,10 @@ def GetDirectoryLicences(root_dir):
     for sub_dir in subdirs:
       full_path = os.path.join(dir_name, sub_dir)
       relative_path = os.path.relpath(full_path, root_dir)
-      ok = True
+      include_directory = True
       for ignored_directory in IGNORED_DIRECTORIES:
-        ok &= relative_path.startswith(ignored_directory)
-      if ok:
+        include_directory &= relative_path.startswith(ignored_directory)
```
Comment (peternewman) [inline tests/scripts/enforce_licence.py]: Shouldn't this be !?

**Comment by peternewman** [comment]
Merge this into OLA too when it goes in.

File: tests/scripts/enforce_licence.py (hunk 1 above)
Comment (nomis52) [inline tests/scripts/enforce_licence.py]: No, that's correct as is.

Comment (peternewman) [inline tests/scripts/enforce_licence.py]: Okay, I'm confused, looking at the code it seems in pseudo code, for each subdir for each ignored dir, check if it matches all the ignored dirs, if it does, add it to the array.

Comment (nomis52) [inline tests/scripts/enforce_licence.py]: Ah, you're correct. PR opened

Comment (peternewman) [inline tests/scripts/enforce_licence.py]: :)

**Pull request merged**